scala/src/com/google/idea/blaze/scala/run/producers/ScalaTestContextProvider.java

```diff
@@ -45,25 +45,41 @@ class ScalaTestContextProvider implements TestContextProvider {
   @Nullable
   @Override
   public RunConfigurationContext getTestContext(ConfigurationContext context) {
-    ScClass testClass = getTestClass(context);
-    if (testClass == null) {
+    ClassAndTestName classAndTestName = getTestClass(context);
+
+    if (classAndTestName == null) {
       return null;
     }
+
+    ScTypeDefinition testClass = classAndTestName.testClass;
+
     ListenableFuture<TargetInfo> target =
         TestTargetHeuristic.targetFutureForPsiElement(
             testClass, TestSizeFinder.getTestSize(testClass));
     if (target == null) {
       return null;
     }
+
     return TestContext.builder(testClass, ExecutorType.DEBUG_SUPPORTED_TYPES)
         .setTarget(target)
-        .setTestFilter(getTestFilter(testClass))
+        .addBlazeFlagsModification(new ScalatestTestSelectorFlagsModification(testClass.getQualifiedName(), classAndTestName.testName))
         .setDescription(testClass.getName())
         .build();
   }
 
+  private static class ClassAndTestName {
+    private final ScTypeDefinition testClass;
+    @Nullable
+    private final String testName;
+
+    public ClassAndTestName(ScTypeDefinition testClass, @Nullable String testName) {
+      this.testClass = testClass;
+      this.testName = testName;
+    }
+  }
+
   @Nullable
-  private static ScClass getTestClass(ConfigurationContext context) {
+  private static ClassAndTestName getTestClass(ConfigurationContext context) {
```

Comment:
Maybe renaming this method to getClassAndTestName()

Reply:
Will do.

Reply:
Did this